### PR TITLE
Admin view all coupon

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/75150

## WHAT (optional)
- add feature admin view all coupons

## HOW
- add feature admin view all coupons

## Evidence (Screenshot or Video)
Screenshot from 2024-05-10 11-19-25
Screenshot from 2024-05-10 11-18-58
Screenshot from 2024-05-10 11-11-03

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/services/coupon.service.ts [hunk 1]
```diff
@@ -13,3 +14,40 @@ export const findCouponByName = async (name: string) => {
...
+  }
+
+  if (data.startDate) {
+    query.andWhere('coupon.startDate <= :startDate', {
+      startDate: dayjs(data.startDate).startOf('day').toDate(),
+    });
+  }
+
+  if (data.endDate) {
+    query.andWhere('coupon.endDate >= :endDate', {
```
Comment (huongnt-2545) [inline src/services/coupon.service.ts]: chỗ này có bị ngược ko nhỉ?
tìm các coupon nằm trong khoảng start_date, end_date

Comment (ghost) [reply src/services/coupon.service.ts]: em có sửa lại rồi ạ

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**